x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -643,123 +447,97 @@ export class ObservabilityAIAssistantClient {
       signal: AbortSignal;
       simulateFunctionCalling?: boolean;
     }
-  ): Promise<Observable<ChatCompletionChunkEvent | TokenCountEvent>> => {
-    const span = apm.startSpan(`chat ${name}`);
-
-    const spanId = (span?.ids['span.id'] || '').substring(0, 6);
-
-    const loggerPrefix = `${name}${spanId ? ` (${spanId})` : ''}`;
-
-    try {
-      const connector = await this.dependencies.actionsClient.get({
-        id: connectorId,
-      });
-
-      let adapter: LlmApiAdapter;
-
-      this.dependencies.logger.debug(`Creating "${connector.actionTypeId}" adapter`);
-
-      switch (connector.actionTypeId) {
-        case ObservabilityAIAssistantConnectorType.OpenAI:
-          adapter = createOpenAiAdapter({
-            messages,
-            functions,
-            functionCall,
-            logger: this.dependencies.logger,
-            simulateFunctionCalling,
-          });
-          break;
-
-        case ObservabilityAIAssistantConnectorType.Bedrock:
-          adapter = createBedrockClaudeAdapter({
-            messages,
-            functions,
-            functionCall,
-            logger: this.dependencies.logger,
-          });
-          break;
-
-        default:
-          throw new Error(`Connector type is not supported: ${connector.actionTypeId}`);
-      }
-
-      const subAction = adapter.getSubAction();
-
-      this.dependencies.logger.debug(`${loggerPrefix}: Sending conversation to connector`);
-      this.dependencies.logger.trace(
-        `${loggerPrefix}:\n${JSON.stringify(subAction.subActionParams, null, 2)}`
-      );
-
-      const now = performance.now();
-
-      const executeResult = await this.dependencies.actionsClient.execute({
-        actionId: connectorId,
-        params: subAction,
-      });
-
-      this.dependencies.logger.debug(
-        `${loggerPrefix}: Received action client response: ${
-          executeResult.status
-        } (took: ${Math.round(performance.now() - now)}ms)`
-      );
-
-      if (executeResult.status === 'error' && executeResult?.serviceMessage) {
-        const tokenLimitRegex =
-          /This model's maximum context length is (\d+) tokens\. However, your messages resulted in (\d+) tokens/g;
-        const tokenLimitRegexResult = tokenLimitRegex.exec(executeResult.serviceMessage);
+  ): Observable<ChatCompletionChunkEvent | TokenCountEvent> => {
+    return of(undefined).pipe(
+      switchMap(() =>
+        from(
+          withAssistantSpan('get_connector', () =>
+            this.dependencies.actionsClient.get({ id: connectorId, throwIfSystemAction: true })
+          )
```

Comment:
Should this be using `guardAgainstInvalidConnector`? And should `guardAgainstInvalidConnector` be calling `withAssistantSpan`?

Reply:
good point, will address